### ISSUE TITLE
"dist" command crashes with "java.lang.StringIndexOutOfBoundsException: String index out of range: -2"

### ISSUE THREAD

**Issue by bwbecker** [playframework/playframework#4091]
I'm developing a project with several subprojects.  The basic project organization is based on the Activator template at https://github.com/adrianhurt/play-multidomain-seed.

I just tried to do the "dist" command for the first time and received 

```
info] 
[info] Your package is ready in /Users/bwbecker/oat/src/oat3/modules/wapp/target/universal/oat3-wapp-3.0-SNAPSHOT.zip
[info] 
[info] 
[info] Your package is ready in /Users/bwbecker/oat/src/oat3/target/universal/oat3-3.0-SNAPSHOT.zip
[info] 
[trace] Stack trace suppressed: run last wsvc/universal:mappings for the full output.
[error] (wsvc/universal:mappings) java.lang.StringIndexOutOfBoundsException: String index out of range: -2
[error] Total time: 6 s, completed 17-Mar-2015 11:44:06 AM
```

Running "last wsvc/universal:mappings" yields

```
[oat3-wsvc] $ last wsvc/universal:mappings
java.lang.StringIndexOutOfBoundsException: String index out of range: -2
	at java.lang.String.substring(String.java:1875)
	at play.PlaySettings$$anonfun$defaultSettings$75$$anonfun$apply$5.apply(PlaySettings.scala:288)
	at play.PlaySettings$$anonfun$defaultSettings$75$$anonfun$apply$5.apply(PlaySettings.scala:287)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
```

Looking at the source to PlaySettings.scala, lines 283-290 are
```scala
    mappings in Universal ++= {
      val confDirectoryLen = confDirectory.value.getCanonicalPath.length
      val pathFinder = confDirectory.value ** ("*" -- "routes")
      pathFinder.get map {
        confFile: File =>
          confFile -> ("conf/" + confFile.getCanonicalPath.substring(confDirectoryLen))
      }
    },
```

So it seems to be related to configuration files.  Some of my subproject config files are soft-linked.  The following set breaks:

```
-rw-r--r--  1 bwbecker  staff   290B  6 Mar 12:17 application.conf
lrwxr-xr-x  1 bwbecker  staff    24B 17 Mar 12:06 local.conf -> ../../../conf/local.conf
-rw-r--r--  1 bwbecker  staff   419B  6 Mar 12:17 prod.conf
lrwxr-xr-x  1 bwbecker  staff    29B  6 Mar 12:17 shared.dev.conf -> ../../../conf/shared.dev.conf
lrwxr-xr-x  1 bwbecker  staff    30B  6 Mar 12:17 shared.prod.conf -> ../../../conf/shared.prod.conf
-rw-r--r--  1 bwbecker  staff   426B  6 Mar 12:17 wsvc.conf
-rw-r--r--  1 bwbecker  staff   4.0K 17 Mar 09:45 wsvc.routes
```

while the following work (the only change is renaming local.conf to shared.local.conf.):

```
-rw-r--r--  1 bwbecker  staff   290B  6 Mar 12:17 application.conf
-rw-r--r--  1 bwbecker  staff   419B  6 Mar 12:17 prod.conf
lrwxr-xr-x  1 bwbecker  staff    29B  6 Mar 12:17 shared.dev.conf -> ../../../conf/shared.dev.conf
lrwxr-xr-x  1 bwbecker  staff    31B 17 Mar 12:10 shared.local.conf -> ../../../conf/shared.local.conf
lrwxr-xr-x  1 bwbecker  staff    30B  6 Mar 12:17 shared.prod.conf -> ../../../conf/shared.prod.conf
-rw-r--r--  1 bwbecker  staff   426B  6 Mar 12:17 wsvc.conf
-rw-r--r--  1 bwbecker  staff   4.0K 17 Mar 09:45 wsvc.routes
```

The full path for the above conf directory is `/Users/bwbecker/oat/src/oat3/modules/wsvc/conf/`

**Comment by benmccann** [comment]
I think this line also probably breaks if you try to disable the PlayLayoutPlugin. We should fix both issues at once
https://groups.google.com/d/msg/play-framework-dev/25rqA8Sq33w/FMsIXb2iTcQJ

**Comment by jroper** [comment]
This should be using the sbt `pair` method, that would solve this.

**Comment by benmccann** [comment]
Docs for `pair`: http://www.scala-sbt.org/0.13/docs/Mapping-Files.html

**Comment by benmccann** [comment]
We should probably first look at https://github.com/playframework/playframework/issues/3473, which is assigned to the 2.4.0 milestone and in which some people have suggested removing the buggy line altogether

**Comment by jroper** [comment]
This will be fixed by #4116.
